index.js

```diff
@@ -97,3 +157,50 @@ function simpleFeature (geom, properties) {
     properties: properties
   };
 }
+
+function simplifyLine (coordinates, tolerance, highQuality) {
+  var simplifiedCoordinates = [];
+  var pts = coordinates.map(function(coord) {
+    return {x: coord[0], y: coord[1]};
+  });
+  simplifiedCoordinates = simplify(pts, tolerance, highQuality).map(function(coords) {
+    return [coords.x, coords.y];
+  });
+
+  return simplifiedCoordinates;
+}
+
+function simplifyPolygon (coordinates, tolerance, highQuality) {
+  var simplifiedCoordinates = [];
+  coordinates.forEach(function(ring) {
+    var pts = ring.map(function(coord) {
+      return {x: coord[0], y: coord[1]};
+    });
+    if (pts.length < 4) {
+      throw new Error('Invalid polygon');
+    }
+    var simpleRing = simplify(pts, tolerance, highQuality).map(function(coords) {
+      return [coords.x, coords.y];
+    });
+    //remove 1 percent of tolerance until enough points to make a triangle
+    while (!checkTriangle(simpleRing)) {
+      tolerance -= tolerance * 0.01;
+      simpleRing = simplify(pts, tolerance, highQuality).map(function(coords) {
+        return [coords.x, coords.y];
+      });
+    }
+    if (simpleRing.length === 3) {
```

Comment:
Because of the triangle checks above, L192-197 could be removed: 

```
// Not needed, because on line 198, we're ensuring the first and last element are equal
if (simpleRing.length === 3) {
      simpleRing.push(simpleRing[0]);
}

// Not needed because you're already handling shapes that are smaller than a triangle above
for (var i = 0; i < 4; i++) {
  if (!simpleRing[i]) simpleRing.push(simpleRing[0]);
}
```